src/SecurityManager.ts

```diff
@@ -72,7 +77,9 @@ async function confirmToDismiss() {
     return !sure;
 }
 
-function makeInputToKey(keyInfo) {
+function makeInputToKey(
+    keyInfo: ISecretStorageKeyInfo,
+): ({ passphrase, recoveryKey }: { passphrase: string, recoveryKey: string }) => Promise<Uint8Array> {
```

Comment:
Should be able to name this without freaking out the compiler
```suggestion
): (keyParams: { passphrase: string, recoveryKey: string }) => Promise<Uint8Array> {
```

Reply:
Aha, indeed... Is there generally a "good" way to type object destructuring, as in [`getSecretStorageKey`](https://github.com/matrix-org/matrix-react-sdk/pull/5307/files/2698a12a6fc61e8e9a1d992b7054f116d355651f#diff-e94b1d35af05f12c5eaf5630b5d4fe9dR97)...? It's quite verbose at the moment, but I think both sides are needs for the destructuring and typing to both take effect...?